### PR TITLE
fix: SDO numbering

### PR DESCRIPTION
**Pull request by Gogetter**
### JIRA link (if applicable) ###

[FPLA-1340](https://tools.hmcts.net/jira/browse/FPLA-1340)

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionService.java [hunk 1]
```diff
@@ -179,6 +180,7 @@ public void addEmptyDirectionsForAssigneeNotInMap(Map<DirectionAssignee, List<El
         AtomicInteger at = new AtomicInteger(2);
 
         return directions.stream()
+            .filter(direction -> YesNo.YES.getValue().equals(direction.getValue().getDirectionNeeded()))
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionService.java]: Static import, same for other changes as well?

**Comment by swalker125** [comment]
I would maybe suggest a new test that's something along the lines of, shouldApplyCorrectNumberingWhenDirectionsAreMarkedAsRemoved

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java [hunk 2]
```diff
@@ -157,6 +158,23 @@ void constructDirectionForCCD_shouldConstructDirectionFromConfigurationAsExpecte
             .build());
     }
 
+    @Test
+    void numberDirections_shouldApplyCorrectNumberingWhenDirectionsAreMarkedAsRemoved() {
+        CaseData caseData = populateCaseDataWithSomeDirectionsMarkedAsRemoved().build();
+
+        List<Element<Direction>> directions = service.combineAllDirections(caseData);
```
Comment (swalker125) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java]: this step seems odd. Why don't you just have a method that populates a list of directions directly. It worries me using a different method to the one under test. Unit tests should exist entirely separate.

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java [hunk 3]
```diff
@@ -157,6 +158,23 @@ void constructDirectionForCCD_shouldConstructDirectionFromConfigurationAsExpecte
...
+
+        List<String> numberedDirectionTypes = service.numberDirections(directions).stream()
+            .map(direction -> direction.getValue().getDirectionType())
+            .collect(toList());
+
+        List<String> expectedDirectionsTypes = IntStream.range(0, numberedDirectionTypes.size())
```
Comment (swalker125) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java]:
```suggestion
        List<String> expectedDirectionTypes = IntStream.range(0, numberedDirectionTypes.size())
```

Comment (swalker125) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java]: this expected direction seems quite weak to me as well. I don't like logic to work out expected values as we're introducing potential failure points. Maybe this is fine however. I shall leave it with you

Comment (swalker125) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java]: I think I would prefer asserting the whole direction so I can see titles AND the removed things 🤔

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java [hunk 4]
```diff
@@ -459,4 +486,14 @@ void shouldNotReturnResponseWhenNoResponseExists() {
                 .custom("Yes")
                 .build()));
     }
+
+    private CaseData.CaseDataBuilder populateCaseDataWithSomeDirectionsMarkedAsRemoved() {
```
Comment (swalker125) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java]:
```suggestion
    private CaseData.CaseDataBuilder caseDataWithDirectionsMarkedAsRemoved() {
```

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java [hunk 5]
```diff
@@ -157,6 +158,23 @@ void constructDirectionForCCD_shouldConstructDirectionFromConfigurationAsExpecte
             .build());
     }
 
+    @Test
+    void numberDirections_shouldApplyCorrectNumberingWhenDirectionsAreMarkedAsRemoved() {
```
Comment (swalker125) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java]: I guess this is actually when directions are marked as not needed

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionService.java [hunk 6]
```diff
@@ -29,6 +29,7 @@
 import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.LOCAL_AUTHORITY;
 import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.OTHERS;
 import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.PARENTS_AND_RESPONDENTS;
+import static uk.gov.hmcts.reform.fpl.enums.YesNo.YES;
```
Comment (swalker125) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionService.java]: hmm this bloats this PR a little 😕

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java (hunk 3 above)
Comment (toireasam) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java]: would it be worth renaming 'x' to something meaningful here like directionNumber?

Comment (Gogetter) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java]: updated

**Comment by toireasam** [comment]
one small comment I left addressed - LGTM

**Pull request merged**